File: push/src/main/java/com/xiaomi/network/LogFallbackAspect.java
package com.xiaomi.network;

import com.elvishew.xlog.Logger;
import com.elvishew.xlog.XLog;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import org.aspectj.lang.JoinPoint;

public class LogFallbackAspect {
    private static final String TAG = LogFallbackAspect.class.getSimpleName();
    private static final Logger logger = XLog.tag(TAG).build();

    public void logFallback(final JoinPoint joinPoint, Fallback fallback, boolean usePort) {
        logger.d(joinPoint.getSignature());
        Gson gson = new GsonBuilder()
                .disableHtmlEscaping()
                .setPrettyPrinting()
                .create();
        logger.d("Fallback " + gson.toJsonTree(fallback));
    }

}

File: push/src/main/java/com/xiaomi/network/LogFallbackAspect.kt
package com.xiaomi.network

import com.elvishew.xlog.Logger
import com.elvishew.xlog.XLog
import com.google.gson.GsonBuilder
import org.aspectj.lang.JoinPoint

class LogFallbackAspect {
    fun logFallback(joinPoint: JoinPoint, fallback: Fallback, usePort: Boolean) {
        logger.d(joinPoint.signature)
        val gson = GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create()
        logger.d("Fallback " + gson.toJsonTree(fallback))
    }

    companion object {
        private val TAG: String = LogFallbackAspect::class.java.simpleName
        private val logger: Logger = XLog.tag(TAG).build()
    }
}
